### PR TITLE
[Ready for Review] Module 1 Update

### PR DESCRIPTION
**Pull request by helendduncan**
## General (applicable to every module)

- [x] Make each module independent of earlier ones (ie. do not rely on examples/specific code from earlier modules)
- [x] Ensure that each module has some small exercises (say 1x 5 min exercise per 45 mins or teaching); add a larger (optional) exercise per course for attendees to complete in their own time
- [ ] Update to use more modern libraries and tools (eg. pandas/scipy/numba/poetry)
- [x] Add instructions to the notebooks that you should "Take 5 minutes to do this exercise"
- [ ] Add estimated timing information at the top of each notebook
- [ ] Switch to a colourblind-friendly colour scheme throughout
- [ ] What do *you* wish people had told you about Python/coding in general?

## Module 1: Introduction to Python

- [x] Set up a virtual environment for the whole course and teach what this is #114 #110 
- [ ] Give a brief introduction about what research software engineering means and why it's important
- [ ] Skip the first few notebooks with the complicated long example
- [ ]   **Idea:** Maybe making `01_01_data_analysis_example` an exercise for Module 2 could be an option? (#85). But there are a few exercises in that module already and the original idea of 01_01 was as a "this is the kind of thing you can do" demo before teaching.
- [ ] Explain difference between a Python file; Jupyter notebook; Python console; what is a terminal?
- [ ]   **Idea:** Make this part of Module 0 (#83)?

### CODE REVIEW

File: module01_introduction_to_python/01_10_classroom_exercises.ipynb [hunk 1]
```diff
@@ -0,0 +1,145 @@
+{
```
Comment (jack89roberts) [inline module01_introduction_to_python/01_10_classroom_exercises.ipynb:1]: <p>typo?</p><p> </p><blockquote>function is in the <code>math</code> library m</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/alan-turing-institute/rse-course/pull/115/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: solutions/module_01.ipynb [hunk 2]
```diff
@@ -2,16 +2,303 @@
  "cells": [
```
Comment (jack89roberts) [inline solutions/module_01.ipynb:2]: <pre class="ql-syntax" spellcheck="false">Line #6.    print("House can fit {} people, and currently has: {}.".format(capacity, occupancy))
</pre><p>Maybe use and link to f-string syntax instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/alan-turing-institute/rse-course/pull/115/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Review by jack89roberts — APPROVED** [review]

File: module01_introduction_to_python/01_10_classroom_exercises.ipynb (hunk 1 above)
Comment (helendduncan) [reply module01_introduction_to_python/01_10_classroom_exercises.ipynb:1]: <p>Fixed</p>

File: solutions/module_01.ipynb (hunk 2 above)
Comment (helendduncan) [reply solutions/module_01.ipynb:2]: <p>Good idea. Done</p>

**Pull request merged**